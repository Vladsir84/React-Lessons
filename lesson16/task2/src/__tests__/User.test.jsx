import React from 'react';
import { shallow } from 'enzyme';
import User from '../User';
import { getUsersList } from '../usersGateway';

jest.mock('../usersGateway', () => {
  return {
    getUsersList: jest.fn(() => Promise.resolve())
  };
});


describe('User', () => {
  it('should should not render', () => {
    const wrappedComponent = shallow(<User />);
    expect(wrappedComponent.find('.user').exists()).toEqual(false);
  });

  it('should render', () => {
    const wrappedComponent = shallow(<User />);
    wrappedComponent.setState({
      userData: {
        avatar_url: 'https://example.com',
        name: 'John',
        location: 'Planet Earth',
      }
    })
    expect(wrappedComponent.find('.user')).toMatchSnapshot(true);
  });

  it('should render', () => {
    shallow(<User />);
    expect(getUsersList).toBeCalled();
  });
});