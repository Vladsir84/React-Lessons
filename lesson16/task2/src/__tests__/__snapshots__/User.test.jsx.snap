// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render 1`] = `
<div
  className="user"
>
  <img
    alt="User Avatar"
    className="user__avatar"
    src="https://example.com"
  />
  <div
    className="user__info"
  >
    <span
      className="user__name"
    >
      John
    </span>
    <span
      className="user__location"
    >
      Planet Earth
    </span>
  </div>
</div>
`;
